### PR TITLE
JAMES-2475 Fix deprecation warnings from dependencies

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: server/data/data-jpa/src/main/java/org/apache/james/user/jpa/model/JPAUser.java [hunk 1]
```diff
@@ -72,6 +72,7 @@ private static PasswordHashFunction chooseHashFunction(String nullableAlgorithm)
         }
     }
 
+    @SuppressWarnings("deprecation")
```
[the remark this answers is not in the document]
Comment (rouazana) [reply server/data/data-jpa/src/main/java/org/apache/james/user/jpa/model/JPAUser.java]: Seems fine for me, we should first document that these hashes should be deprecated (if not already done).

**Comment by mbaechler** [comment]
could you please explain why you choose icu4j instead of guava CharMatcher ?

**Comment by aduprat** [comment]
> could you please explain why you choose icu4j instead of guava CharMatcher ?

_CharMatcher.javaLetterOrDigit()_ is deprecated in Guava & they recommend to use that library instead (the main purpose is to support supplementary characters.)
https://google.github.io/guava/releases/25.1-jre/api/docs/

**Comment by trantienduchn** [comment]
Again :)
```
[b2f634574859e2d30000cdd803fb5b8e07d9051f] [ERROR] Failures: 
[b2f634574859e2d30000cdd803fb5b8e07d9051f] [ERROR]   ESReporterTest.esMetricReporterShouldProduceDocumentsOnAnElasticsearchContainer Multiple Failures (2 failures)
[b2f634574859e2d30000cdd803fb5b8e07d9051f] 	Mapped port can only be obtained after the container is started
[b2f634574859e2d30000cdd803fb5b8e07d9051f] 	<no message> in java.lang.NullPointerException
[b2f634574859e2d30000cdd803fb5b8e07d9051f] [ERROR]   ESReporterTest.esMetricReporterShouldProduceDocumentsOnAnElasticsearchContainerWhenRecordingTimeMetric Multiple Failures (2 failures)
[b2f634574859e2d30000cdd803fb5b8e07d9051f] 	Mapped port can only be obtained after the container is started
[b2f634574859e2d30000cdd803fb5b8e07d9051f] 	<no message> in java.lang.NullPointerException
[b2f634574859e2d30000cdd803fb5b8e07d9051f] [INFO] 
[b2f634574859e2d30000cdd803fb5b8e07d9051f] [ERROR] Tests run: 8, Failures: 2, Errors: 0, Skipped: 0
```

**Pull request closed without merging**